lib/rubocop/cop/rails/where_equals.rb

```diff
@@ -88,15 +93,20 @@ def extract_column_and_value(template_node, value_node)
           [Regexp.last_match(1), value]
         end
 
-        def build_good_method(column, value)
+        def build_good_method(method_name, column, value)
           if column.include?('.')
             table, column = column.split('.')
 
-            "where(#{table}: { #{column}: #{value} })"
+            "#{method_name}(#{table}: { #{column}: #{value} })"
           else
-            "where(#{column}: #{value})"
+            "#{method_name}(#{column}: #{value})"
           end
         end
+
+        def where_not?(node)
+          receiver = node.receiver
+          receiver&.send_type? && receiver&.method?(:where)
```

Comment:
Can you tweak to the following?

```suggestion
          return false unless (receiver = node.receiver)

          receiver.send_type? && receiver.method?(:where)
```